src/libs/Violations/ViolationsUtils.ts

```diff
@@ -0,0 +1,69 @@
+import reject from 'lodash/reject';
+import Onyx from 'react-native-onyx';
+import ONYXKEYS from '@src/ONYXKEYS';
+import {PolicyCategories, PolicyTags, Transaction, TransactionViolation} from '@src/types/onyx';
+import possibleViolationsByField, {ViolationField} from './possibleViolationsByField';
+
+const ViolationsUtils = {
+    getViolationForField(transactionViolations: TransactionViolation[], field: ViolationField, translate: (key: string) => string): string[] {
+        return transactionViolations.filter((violation) => possibleViolationsByField[field]?.includes(violation.name)).map((violation) => translate(violation.name));
+    },
+
+    getViolationsOnyxData(
+        /** The transaction to check for policy violations. */
+        transaction: Transaction,
+        /** An array of existing transaction violations. */
+        transactionViolations: TransactionViolation[],
+        /** Indicates if the policy requires tags. */
+        policyRequiresTags: boolean,
+        /** Collection of policy tags and their enabled states. */
+        policyTags: PolicyTags,
+        /** Indicates if the policy requires categories. */
+        policyRequiresCategories: boolean,
+        /** Collection of policy categories and their enabled states. */
+        policyCategories: PolicyCategories,
+    ): {
+        onyxMethod: string;
+        key: string;
+        value: TransactionViolation[];
```

Comment:
NAB but I wonder if it's worth defining this as a new OnyxUpdate type. Could be worth discussing in #expensify-open-source, what do you think?

Reply:
I don't have enough of an understanding of Onxy to have a strong opinion on it. That sounds like a good idea to me at first glance as I'm in favour of better typing in general.